include/bluetooth/services/mds.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef MDS_H_
+#define MDS_H_
+
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+
+/**@file
+ * @defgroup bt_mds Memfault Diagnostic GATT service API
+ * @{
+ * @brief API for the Memfault Diagnostic GATT service (MDS).
+ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @brief UUID of the Memfault Diagnostic Service. */
+#define BT_UUID_MDS_VAL \
+	BT_UUID_128_ENCODE(0x54220000, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+/** @brief UUID of the MDS Supported feature Characteristic. */
+#define BT_UUID_MDS_SUPPORTED_FEATURES_VAL \
+	BT_UUID_128_ENCODE(0x54220001, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+/** @brief UUID of the MDS Device Identifier Characteristic. */
+#define BT_UUID_MDS_DEVICE_IDENTIFIER_VAL \
+	BT_UUID_128_ENCODE(0x54220002, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+/** @brief UUID of the MDS Data URI Characteristic. */
+#define BT_UUID_MDS_DATA_URI_VAL \
+	BT_UUID_128_ENCODE(0x54220003, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+/** @brief UUID of the MDS Authorization Characteristic. */
+#define BT_UUID_MDS_AUTHORIZATION_VAL \
+	BT_UUID_128_ENCODE(0x54220004, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+/** @brief UUID of the MDS Data Export Characteristic. */
+#define BT_UUID_MDS_DATA_EXPORT_VAL \
+	BT_UUID_128_ENCODE(0x54220005, 0xf6a5, 0x4007, 0xa371, 0x722f4ebd8436)
+
+#define BT_UUID_MEMFAULT_DIAG          BT_UUID_DECLARE_128(BT_UUID_MDS_VAL)
+#define BT_UUID_MDS_SUPPORTED_FEATURES BT_UUID_DECLARE_128(BT_UUID_MDS_SUPPORTED_FEATURES_VAL)
+#define BT_UUID_MDS_DEVICE_IDENTIFIER  BT_UUID_DECLARE_128(BT_UUID_MDS_DEVICE_IDENTIFIER_VAL)
+#define BT_UUID_MDS_DATA_URI           BT_UUID_DECLARE_128(BT_UUID_MDS_DATA_URI_VAL)
+#define BT_UUID_MDS_AUTHORIZATION      BT_UUID_DECLARE_128(BT_UUID_MDS_AUTHORIZATION_VAL)
+#define BT_UUID_MDS_DATA_EXPORT        BT_UUID_DECLARE_128(BT_UUID_MDS_DATA_EXPORT_VAL)
+
+/** @brief Memfault Diagnostic Service callback structure.
+ */
+struct bt_mds_cb {
+	/** @brief Memfault access enable callback.
+	 *
+	 * If this callback in not implemented then the all peers can access the Memfault data.
+	 * The Memfault Diagnostic service allows only one client subscription.
+	 *
+	 * @param[in] conn Connection object.
+	 *
+	 * @retval True if peer associated with connection object has granted access to
+	 *         the Memfault data.
+	 *	   False if access to the Memfault data is forbidden.
+	 */
+	bool (*access_enable)(struct bt_conn *conn);
```

Comment:
Should we consider informing which data is being accessed (or at least if we agree on data read or subscription enabling)? I think this information may actually be useful.

Reply:
Added